### PR TITLE
Fix: CMSB deprecated API usage in MetricEventPublisher

### PR DESCRIPTION
**Pull request by Abijithkrishna**
**What**:
A deprecated constructor was being used in MetricEventPublisher.

**Why**:
A warning would be logged when building, referring to the use of deprecated methods. If left unchecked a compilation issue might occur in the future.

**How**:
The constructor was replaced by the inner workings of it, (ie) the constructor was internally calling another constructor which has been called directly. 

**Checklist**:

- [N/A] Documentation added
      
- [ x] Changelog updated
      
- [N/A] Tests added
      
- [ x] Ready to be merged

### CODE REVIEW

File: chaos-monkey-docs/src/main/asciidoc/changes.adoc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [[changes]]
 == Changes in {project-version}
-
+https://github.com/codecentric/chaos-monkey-spring-boot/pull/131 Deprecated constructor usage removal
```
Comment (maiksensi) [inline chaos-monkey-docs/src/main/asciidoc/changes.adoc]: Please us a markdown link here like: `- https://github.com/codecentric/chaos-monkey-spring-boot/pull/131[#131] Update dependencies to latest stable versions.` Please note, that this is a list!

File: chaos-monkey-docs/src/main/asciidoc/changes.adoc [hunk 2]
```diff
@@ -10,4 +10,6 @@
 === Contributors
 This release was only possible because of these great humans:
 
+https://github.com/Abijithkrishna
```
Comment (maiksensi) [inline chaos-monkey-docs/src/main/asciidoc/changes.adoc]: Please use a markdown link here (also, this should be a list) like
`- https://github.com/Abijithkrishna[@Abijithkrishna]`.

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/component/MetricEventPublisher.java [hunk 3]
```diff
@@ -25,7 +25,8 @@ public void publishMetricEvent(MetricType metricType, String... tags) {
     }
 
     public void publishMetricEvent(MetricType metricType, AtomicInteger atomicTimeoutGauge) {
-        MetricEvent metricEvent = new MetricEvent(this, metricType, atomicTimeoutGauge);
+        //MetricEvent constructor has been deprecated. So falling back to the internally used Constructor
```
Comment (maiksensi) [inline chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/component/MetricEventPublisher.java]: I noticed that all constructors are public in `MetricEvent` and this not "internally". I think we can remove that comment if that's what you wanted to point out here. If not, please elaborate a bit on what the comment explains.

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/component/MetricEventPublisher.java [hunk 4]
```diff
@@ -25,7 +25,8 @@ public void publishMetricEvent(MetricType metricType, String... tags) {
...
+        MetricEvent metricEvent = new MetricEvent(this, metricType, atomicTimeoutGauge!=null? atomicTimeoutGauge.longValue() : -1,null);
```
Comment (maiksensi) [inline chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/component/MetricEventPublisher.java]: ❓ Does it make sense to extract the conditional to a separate variable for better readability?
❓ Any change you can run a formatter on the code? We tend to use IntelliJ's standard formatter (it is not yet implemented in the CI pipeline but we will change that soon, so this ain't a manual process anymore).

**Comment by maiksensi** [comment]
Hey @Abijithkrishna,

this evening we hosted a Hackergarten (context: hackergarten.net) and one of the issues tackled by a participant was removing the deprecated constructor. I hope you don't mind. If you want to, add yourself to the contributors list (`changes.adoc`) (using a MR) for this release.

**Comment by Abijithkrishna** [comment]
Sorry was not available until today, thats why I couldn't make the changes you had requested. Glad someone else stepped forward.

**Pull request closed without merging**